### ISSUE TITLE
LOG_LEVEL not respected

### ISSUE THREAD

**Issue by oers** [Azure/azure-storage-fuse#652]
### Which version of the blobfuse was used?
1.4.1

### Which OS (please include version) are you using?
Ubuntu 18.04.4 LTS

### What problem was encountered?
Blobfuse logs excessiveley to syslog. I started with WARNING. This logged every Request fully. ERROR reduced this a bit. OFF still logs (see image).

![grafik](https://user-images.githubusercontent.com/1115563/135211349-b347366f-139f-4580-abdb-8ce88f341960.png)

### Have you found a mitigation/solution?
No.

### If relevant, please share your mount command.
`
[Unit]
Description=A virtual file system adapter for Azure Blob storage.
After=network-online.target
Requires=network-online.target

[Service]

# Configures the mountpoint.<path/to/the/mounting/point>
Environment=BlobMountingPoint=/nexus
# Configures the tmp location for the cache. Always configure the fastest disk (SSD or ramdisk) for best performance.
Environment=BlobTmp=/mnt/blobfusetmp
# Config file path
Environment=BlobConfigFile=/opt/blobfuse-connection.cfg

# Enables HTTPS communication with Blob storage. True by default. HTTPS must be if you are communicating to the Storage Container through OAuth.
Environment=BlobUseHttps=true
# Blobs will be cached in the temp folder for this many seconds. 120 seconds by default. During this time, blobfuse will not check whether the file is up to date or not.
Environment=BlobFileCacheTimeOutInSeconds=120
# Enables logs written to syslog. Set to LOG_WARNING by default. Allowed values are LOG_OFF|LOG_CRIT|LOG_ERR|LOG_WARNING|LOG_INFO|LOG_DEBUG
Environment=BlobLogLevel=LOG_OFF
# Enables attributes of a blob being cached. False by default. (Only available in blobfuse 1.1.0 or above)
Environment=BlobUseAttrCache=false

# Under the hood
Environment=attr_timeout=240
Environment=entry_timeout=240
Environment=negative_timeout=120
Type=simple
#ExecStart=/home/AzureUser/azure-storage-fuse/build/blobfuse /home/AzureUser/mntblobfuse --tmp-path=/home/AzureUser/tmpblobfuse --config-file=/home/AzureUser/azure-storage-fuse/connectionnarasubfuse.cfg --use-https=true --file-cache-timeout-in-seconds=20 --log-level=LOG$
#ExecStop=/usr/bin/fusermount -u /home/AzureUser/mntblobfuse
ExecStart=/usr/bin/blobfuse ${BlobMountingPoint} --log-level=${BlobLogLevel} --tmp-path=${BlobTmp} --config-file=${BlobConfigFile} --use-https=true --file-cache-timeout-in-seconds=20 --use-attr-cache=true -o attr_timeout=120 -o entry_timeout=120 -o negative_timeout=120 $
ExecStop=/usr/bin/fusermount -u ${BlobMountingPoint}

[Install]
WantedBy=multi-user.target

`

**Comment by vibhansa-msft** [comment]
--log-level=LOG_OFF added to mount cli, is turning off the logging at blobfuse.

**Comment by vibhansa-msft** [comment]
LOG_ERR will dump only error logs and that's the minimal you can get.

**Comment by vibhansa-msft** [comment]
Looking at the image that you have shared, it does not appear to be blobfuse logs. These appears to be coming from libfuse library itself. Can you check your mount command and remove "-d" from it, if it exists. "-d" is used to load libfuse in debug mode and foreground thus libfuse will emit heavy logs.

**Comment by oers** [comment]
`ExecStart=/usr/bin/blobfuse ${BlobMountingPoint} --log-level=${BlobLogLevel} --tmp-path=${BlobTmp} --config-file=${BlobConfigFile} --use-https=true --file-cache-timeout-in-seconds=20 --use-attr-cache=true -o attr_timeout=120 -o entry_timeout=120 -o negative_timeout=120 $`

this is the mount command, taken from the system.d service definition provided here https://github.com/Azure/azure-storage-fuse/tree/master/systemd

**Comment by oers** [comment]
Maybe part of the confusion is this file: https://github.com/Azure/azure-storage-fuse/blob/master/systemd/10-blobfuse.conf
the Readme does not mention where I should put this. This may redirect the REST logs?.

**Comment by oers** [comment]
I created the rsylog filter 10-blobfuse.conf and added the logrotate definition. Now the blobfuse.log is growing and the logs no longer appear in syslog.

**Comment by vibhansa-msft** [comment]
The rsyslog filter and logrotate files are for that purpose only. It just redirect blobfuse logs to a seperate file. It will not solve the log volume issue. These logs are not coming from blobfuse code, its coming from libfuse. So may you need to check if you have enabled libfuse logs by any other means.

**Comment by oers** [comment]
After the rsyslog filter the blobfuse.log is growing very fast. This means that the logs come from blobfuse, right?

**Comment by vibhansa-msft** [comment]
logs are getting dumped to that file because the string in log contains "blobfuse" and thats what the filter looks for. As I told earlier these logs belong to libfuse, but libfuse being a library which is loaded inside blobfuse they are coming with blobfuse PID. You need to check what is wrong with the libfuse lib/configs on your system. This does not happen normally. Suspecting some issue with libfuse. Have you installed standard libfuse or you have built a custom one on your local system. Also share contents of your /etc/fuse.conf file.

**Comment by oers** [comment]
```
# /etc/fuse.conf - Configuration file for Filesystem in Userspace (FUSE)

# Set the maximum number of FUSE mounts allowed to non-root users.
# The default is 1000.
#mount_max = 1000

# Allow non-root users to specify the allow_other or allow_root mount options.
#user_allow_other
~

```

libfuse is just installed via apt

```
libfuse2/bionic,now 2.9.7-1ubuntu1 amd64 [installed]
  Filesystem in Userspace (library)
```

**Comment by vibhansa-msft** [comment]
fuse.conf looks clean and version is also correct. If you have any other machine/VM can you try it there with a fresh setup. I suspect some config in linux is triggering libfuse to dump its logs.

**Comment by vibhansa-msft** [comment]
Also can you share one of your sample log file, just want to double check all are libfuse logs only.

**Comment by oers** [comment]
[blobfuse.zip](https://github.com/Azure/azure-storage-fuse/files/7251194/blobfuse.zip)

```
ubuntu-18-04:~$ sysctl kernel.printk
kernel.printk = 4       4       1       7

```

**Comment by vibhansa-msft** [comment]
printk output looks normal only. I checked the syslog file and its all filled up with libfuse logs only, no logs from actual blobfuse code. There is some issue in your environment, libfuse does not emit debug logs unless requested.

**Comment by vibhansa-msft** [comment]
Can you mount blobfuse from a using regular bash command and check behavior remains same or not.

**Comment by oers** [comment]
Could this be some difference in systemd between the types (both examples here have different Type=XXX value)
Type=simple -> captures console output (which may have libfuse logs)
Type=forking -> forks and no extra logs?

I can try to do the old mount approach. But I need a test window for that on the machine. This may take a while.

**Comment by vibhansa-msft** [comment]
Possible. You are trying the one with config file right ?

**Comment by vibhansa-msft** [comment]
I tried both service files on my VM and do not see those libfuse logs getting dumped. You can verify when you get the test machine. I still believe its some system setting causing libfuse to dump all those trace logs.

**Comment by oers** [comment]
Yes I used the config file one. Will try with a test machine. Thanks for the feedback.

**Comment by vibhansa-msft** [comment]
Were you able to get to the root-cause of this. As there are no updates on this, I will close this issue by EOD tommorow.

**Comment by oers** [comment]
I still had no time time to invest into a deeper evaluation. Plan is to try some approaches in November.

**Comment by kesavkolla** [comment]
@vibhansa-msft I am using the with the config file systemd. I am seeing huge debug logs under /var/log/blobfuse.log

If I remove -d from the ExecStart then the mount itself is not happening. Can someone shed light on this debug logs?

**Comment by vibhansa-msft** [comment]
can you share what version you are using ? Earlier we have seen this issue of "-d" making things work and removing it just not let the mount go through due to curl issues. In latest versions we have fixed it. You can try without systemd (like a regular mount command) and see it that works, just to narrow down whether its systemd related issue or general blobfuse related issue.

**Comment by kesavkolla** [comment]
I am using version 1.4.2

```
blobfuse -v
blobfuse 1.4.2
```

I will try using without systemd. 

FYI I just copied the unit file from the repo with config example. The unit file in the repo has -d in the end. What my observation is it does print lots of logs in /var/log/blobfuse.log

**Comment by vibhansa-msft** [comment]
'-d' option is to mount with debug logs enable for libfuse driver also and that will generate a huge volume of logs. Ideally '-d' is given only when there is some serious issue and we need to debug on interaction with kernel fuse module. So removing it is completely safe. I will check why '-d' is given in the sample file. Thanks for bringing this up.
